go-controller/pkg/network-attach-def-controller/network_attach_def_controller.go

```diff
@@ -280,3 +331,28 @@ func nadNeedsUpdate(oldNAD, newNAD *nettypes.NetworkAttachmentDefinition) bool {
 
 	return !reflect.DeepEqual(oldNAD.Spec, newNAD.Spec)
 }
+
+func (nadController *NetAttachDefinitionController) GetActiveNetworkForNamespace(namespace string) (util.NetInfo, error) {
+	if !util.IsNetworkSegmentationSupportEnabled() {
+		return &util.DefaultNetInfo{}, nil
+	}
+	nadController.primaryNetworks.LockKey(namespace)
+	defer nadController.primaryNetworks.UnlockKey(namespace)
+	primaryNetworks, _ := nadController.primaryNetworks.Load(namespace)
+	if len(primaryNetworks) > 1 {
+		return nil, util.NewUnknownActiveNetworkError(namespace)
```

Comment:
Should the UDN CRD prevent this from happening? If so, but NADs are being used directly with no UDN CRD, should we really care about NADs misuse? 
Also, if this happens, why wouldn't we just fail the reconciliation?

Reply:
yeah it shouldnt happen if the user only uses UDN CRs. This check was added by @qinqon I think before we had the UDN CRD. My objective in this PR was to just keep the current get method as is and migrate it to use the controller, so I didn't change this logic. I don't think it hurts to keep it, but either way should probably be addressed in a separate PR.

Reply:
Ok, it's just that it complicates things a bit not just here but also `unsetActiveNetworkForNamespace`.

Reply:
BTW, my second question:  why wouldn't we just fail the reconciliation in this case? Otherwise we would be passing a configuration to network manager that is invalid.

Reply:
agreed. I'll add this